pkg/controller/logstorage/logstorage_controller.go

```diff
@@ -639,25 +687,33 @@ func (r *ReconcileLogStorage) deleteInvalidECKManagedPublicCertSecret(ctx contex
 	return r.client.Delete(ctx, secret)
 }
 
-// getElasticsearchCertificateSecrets retrieves Elasticsearch certificate secrets needed for Elasticsearch to run or for
-// components to communicate with Elasticsearch. The order of the secrets returned are:
-// 1) The certificate secret needed for Elasticsearch (in the operator namespace). If the user didn't create this it is
+// getESGatewayCertificateSecrets retrieves certificate secrets needed for ES Gateway to run or for
+// components to communicate with Elasticsearch/Kibana through ES Gateway. The order of the secrets returned are:
+// 1) The certificate secret needed for ES Gateway (in the operator namespace). If the user didn't create this it is
 //    created.
-// 2) The certificate secret needed for Elasticsearch (in the Elasticsearch namespace).
-// 3) The certificate mounted by other clients that connect to Elasticsearch.
-func (r *ReconcileLogStorage) getElasticsearchCertificateSecrets(ctx context.Context, instl *operatorv1.InstallationSpec) (oprKeyCert *corev1.Secret, esKeyCert *corev1.Secret, certSecret *corev1.Secret, err error) {
+// 2) The certificate secret needed for ES Gateway (in the Elasticsearch namespace).
+// 3) The certificate mounted by other clients that connect to Elasticsearch/Kibana through ES Gateway (in the operator namespace).
+// 4) The certificate mounted by es curator and es metrics to connect to Elasticsearch/Kibana through ES Gateway (in the Elasticsearch namespace).
+func (r *ReconcileLogStorage) getESGatewayCertificateSecrets(ctx context.Context, instl *operatorv1.InstallationSpec) (
```

Comment:
From the looks of it, the newly introduced 4th return variable is redundant. It is used in two places:
- to calculate a hash at a place where the 3rd variable would give an identical hash
- to pass to the elasticsearch_metrics.go, where the secret is already copied to the es namespace

Reply:
@vberezny, extending on this I think you should only need to pass back the operator name spaced secrets, and then just get the Elasticsearch controller to copy those secrets to the Elasticsearch namespace explicitly. 

I think you should also just hash on the secrets put in the elasticsearch namespace (this is an improvement to what we have). We don't want to roll over Elasticsearch when the operator secrets change, it's a little too preemptive Elasticsearch doesn't actually use those secrets. We want Elasticsearch to roll over only when the secrets it actually uses (the copies) change.

As for es metrics, just pass the name of the secret down, or use the hard coded value.